components/brave_sync/brave_sync_prefs.cc

```diff
@@ -102,7 +102,9 @@ std::string Prefs::GetSeedPath() {
   return kSyncV2Seed;
 }
 
-std::string Prefs::GetSeed() const {
+std::string Prefs::GetSeed(bool* failed_to_decrypt) const {
+  if (failed_to_decrypt)
+    *failed_to_decrypt = false;
   const std::string encoded_seed = pref_service_->GetString(kSyncV2Seed);
   std::string encrypted_seed;
   if (!base::Base64Decode(encoded_seed, &encrypted_seed)) {
```

Comment:
If the seed is badly encoded, we should probably say that the decryption failed, no?

Reply:
Yes, `failed_to_decrypt` arg at `BraveSyncServiceImpl::GetOrCreateSyncCode` stops the re-creating of the passphrase when string result is empty, `Base64Decode` failure should be also done in this way.

The reason why I didn't do that initially - I didn't see such failure in a normal conditions, opposite to `OSCrypt::DecryptString` with locked keyring.

Thanks, I will fix this.

Reply:
@fmarier I have made the change for it in https://github.com/brave/brave-core/pull/11935/commits/9204d4a5c261e44183640fefd84a11c21d2c3de1